### PR TITLE
Fix workstation react agent bug

### PR DESCRIPTION
**Pull request by qbc2016**
## Description

1. modify react agent and services in `workflow_node.py`
2. support partial for add function in ServiceToolkit

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has passed all tests
- [ ]  Docstrings have been added/updated in Google Style
- [ ]  Documentation has been updated
- [ ]  Code is ready for review

### CODE REVIEW

File: src/agentscope/service/service_toolkit.py [hunk 1]
```diff
@@ -195,6 +195,10 @@ def bing_search(query: str, api_key: str, num_results=10):
 
         """
 
+        if isinstance(service_func, partial):
```
Comment (DavdGao) [inline src/agentscope/service/service_toolkit.py:198]: Why do we need to consider the partial function here?

Comment (qbc2016) [reply src/agentscope/service/service_toolkit.py:198]: For example, api_key should be passed in bing_serach.
I am not sure if there is another way to implement the following in `workflow_node.py`
 `service_toolkit.add(bing_search, api_key=BING_API_KEY, num_results=3)`

Comment (qbc2016) [reply src/agentscope/service/service_toolkit.py:199]: add TODO here

**Review by DavdGao — APPROVED** [review]
lgtm

**Pull request merged**